wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -166,75 +166,57 @@ private[wallet] trait TransactionProcessing extends WalletLogger {
 
     logger.debug(
       s"Processing transaction=${transaction.txIdBE} with blockHash=$blockHashOpt")
-    for {
-      aggregate <- {
-
-        val incomingTxoFut: Future[Vector[SpendingInfoDb]] =
-          spendingInfoDAO
-            .findTx(transaction)
-            .flatMap {
-              // no existing elements found
-              case Vector() =>
-                processNewIncomingTx(transaction, blockHashOpt, newTags)
-                  .map(_.toVector)
-
-              case txos: Vector[SpendingInfoDb] =>
-                FutureUtil
-                  .sequentially(txos)(txo =>
-                    processExistingIncomingTxo(transaction, blockHashOpt, txo))
-                  .map(_.toVector)
-            }
+    def incomingTxoFut: Future[Vector[SpendingInfoDb]] =
```

Comment:
Let's just move these helper methods out of `processTransactionImpl()` now since they aren't `val` anymore.